lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java

```
@@ -341,6 +359,24 @@ private void generateCode(MethodDefinition methodDefinition) throws IOException
 
       out.println();
 
+      out.println("  /**");
+      out.println("   * Transforms a Java Throwable into either a dataflow-error or a Panic exception.");
+      out.println("   * @param t The Throwable thrown from the {@code execute} method.");
+      out.println("   */");
+      out.println("  private static Object handleThrowable(Throwable t, Node locationNode) {");
+      out.println("    CompilerDirectives.transferToInterpreterAndInvalidate();");
```

I am not sure we should catch and convert all `Throwable`. I know that `ProxyNode` & co. in Truffle instrumentation rely on `ThreadDeath` being propagated for _rewind frame_ functionality in debugger. Also silently swallowing `InterruptedException` might have _uninterruptible_ consequences. We should probably at least propagate `ThreadDeath`.

---

Right. I have learned that the hard way - we should not. I have changed that to only catch some specific exceptions that we can handle at this point. For now, it's only `UnuspportedSpecializationException`. See in c4f197db392ba802dd030490bdb0d299848bf1db